src/sources/redis/mod.rs

```diff
@@ -226,10 +246,28 @@ async fn handle_line(
 
                 let events = events.into_iter().map(|mut event| {
                     if let Event::Log(ref mut log) = event {
-                        log.try_insert(log_schema().source_type_key(), Bytes::from("redis"));
-                        log.try_insert(log_schema().timestamp_key(), now);
-                        if let Some(redis_key) = redis_key {
-                            event.as_mut_log().insert(redis_key, key);
+                        log_namespace.insert_vector_metadata(
+                            log,
+                            log_schema().source_type_key(),
+                            "source_type",
```

Comment:
This is somewhat of a nit - but it sounds like long term we should be using `path!()` on these keys, ex: https://github.com/vectordotdev/vector/pull/14985/files#diff-ce8ca490d9ddbe754a3227df21f669c2f7c95198ca26c9067076ad364617bbcdR559-R560